sdk/keyvault/keyvault/src/keys/index.ts

```diff
@@ -0,0 +1,476 @@
+      exponentialRetryPolicy(
+        retryOptions.retryCount,
+        retryOptions.retryIntervalInMS,
+        RetryConstants.MIN_RETRY_INTERVAL_MS, // Minimum retry interval to prevent frequent retries
+        retryOptions.maxRetryDelayInMs
+      ),
+      redirectPolicy(),
+      signingPolicy(credential)
+    ];
+
+    return {
+      httpClient: pipelineOptions.HTTPClient,
+      httpPipelineLogger: pipelineOptions.logger,
+      requestPolicyFactories
+    };
+  }
+
+  public readonly vaultBaseUrl: string;
+
+  public readonly pipeline: Pipeline;
+
+  protected readonly credential: ServiceClientCredentials;
+  private readonly client: KeyVaultClient;
+
+  /**
+   * Creates an instance of KeysClient.
+   * @param {string} url the base url to the key vault.
+   * @param {ServiceClientCredentials} credential credential.
+   * @param {(Pipeline | NewPipelineOptions)} [pipelineOrOptions={}] Optional. A Pipeline, or options to create a default Pipeline instance.
+   *                                                                 Omitting this parameter to create the default Pipeline instance.
+   * @memberof KeysClient
+   */
+  constructor(
+    url: string,
+    credential: ServiceClientCredentials,
+    pipelineOrOptions: Pipeline | NewPipelineOptions = {}
+  ) {
+    this.vaultBaseUrl = url;
+    this.credential = credential;
+    if (isNewPipelineOptions(pipelineOrOptions)) {
+      this.pipeline = KeysClient.getDefaultPipeline(
+        credential as ServiceClientCredentials,
+        pipelineOrOptions
+      );
+    } else {
+      this.pipeline = pipelineOrOptions;
+    }
+
+    this.client = new KeyVaultClient(credential, this.pipeline);
+  }
+
+  private static getUserAgentString(telemetry?: TelemetryOptions) {
+    const userAgentInfo: string[] = [];
+    if (telemetry) {
+      if (userAgentInfo.indexOf(telemetry.value) === -1) {
+        userAgentInfo.push(telemetry.value);
+      }
+    }
+    const libInfo = `Azure-KeyVault-Keys/${SDK_VERSION}`;
+    if (userAgentInfo.indexOf(libInfo) === -1) {
+      userAgentInfo.push(libInfo);
+    }
+    const defaultUserAgentInfo = getDefaultUserAgentValue();
+    if (userAgentInfo.indexOf(defaultUserAgentInfo) === -1) {
+      userAgentInfo.push(defaultUserAgentInfo);
+    }
+    return userAgentInfo.join(" ");
+  }
+
+  // TODO: do we want Aborter as well?
+
+  /**
+   * The create key operation can be used to create any key type in Azure Key Vault. If the named key
+   * already exists, Azure Key Vault creates a new version of the key. It requires the keys/create
+   * permission.
+   * @summary Creates a new key, stores it, then returns key parameters and attributes to the client.
+   * @param keyName The name of the key.
+   * @param keyType The type of the key.
+   * @param [options] The optional parameters
+   * @returns Promise<Key>
+   */
+  public async createKey(
+    keyName: string,
+    keyType: JsonWebKeyType,
+    options?: CreateKeyOptions
+  ) {
```

Comment:
In other languages we're splitting this out into 3 methods:
createKey(name, keyType, {  keyOps?, enabled?, notBefore?, expires?, tags?, requestOptions? })
createEcKey(name, {  curve?, keyOps?, enabled?, notBefore?, expires?, tags?, requestOptions? })
createRsaKey(name, {  keySize?, keyOps?, enabled?, notBefore?, expires?, tags?, requestOptions? })